### DISCUSSION TITLE
NPM postinstall script fails, when installing projen-based package in non-projen projects

### DISCUSSION THREAD

**Discussion by bweigel** [projen/projen#2492] [Q&A] [answered]
### Intro 

In our company we have an internal cdk-construct library, which is a monorepo creating multiple packages. Kind of like aws-cdk was in version 1. 
Last week I migrated this monorepo from lerna to projen and [NxMonorepoProject](https://aws.github.io/aws-prototyping-sdk/typescript/nx-monorepo/index.html) with nested `AwsCdkConstructLibrary` projects. 

### Problem 

Some of the projects make use of `postinstall` scripts. The postinstall scripts are mainly used to install dependencies needed for lambda functions provided by our libraries. We package the libraries and upload them to private github npm repos. 
Now when other projects try to install these packages (with `postinstall` scripts), the scripts get triggered and try to run `npx projen postinstall` and not the actual script, because projen puts the actual command in `.projen/tasks.json` (which isn't packaged into the `*.tgz`). And this is when the following error happens:

```
npm ERR! path /home/runner/work/***/***/cdk/node_modules/@***/***
npm ERR! command failed
npm ERR! command sh -c npx projen postinstall
npm ERR! npm WARN exec The following package was not found and will be installed: projen@0.67.69
npm ERR! Unknown command: postinstall`,
```

Did anyone run into this before? Is there an escape hatch, such that projen does not delegate the `postinstall` script into the `tasks.json`?

**Comment by mrgrain** [comment, marked as answer]
Hi, that makes sense. I think your problem originates from mixing projen lifecycle with npm lifecycle. And like you've found out they don't mix very well.

I'd recommend that you [setScript()](https://projen.io/api/API.html#setscriptname-command-) to define your `postinstall` scripts and bypass projen completely.

If you need it for both, you'll have to double up the task definition.

**Comment by bweigel** [comment]
Cool thanks for your suggestion. That solves my issue.
